sql/catalyst/src/main/java/org/apache/spark/sql/connector/distributions/OrderedDistribution.java

```diff
@@ -32,4 +32,13 @@
    * Returns ordering expressions.
    */
   SortOrder[] ordering();
+
+  /**
+   * Returns the number of partitions required by this write.
```

Comment:
I'm not sure if `required by this write` is accurate since we could potentially reuse this interface for read as well in future?

Reply:
Hmm... you're right we seem to try to co-use the interface in both read/write path, so it might be confusing if something is only saying about write.

I'm not sure this method can be used in read path as well though. For read path, data source is already able to control the parallelism (say, partitions), so no need to pass the requirement and let Spark does it instead.

So I guess this is a point of divergence between read and write. If we can explicitly and effectively classify the methods as "common", "read specific", "write specific" then that would be great, but let's hear others' voices on this.

Reply:
Agree this is not quite useful for read. Does it make sense to put this in `RequiresDistributionAndOrdering`? it will deviate from the catalyst definition of distribution but this is for write path only.

Reply:
That was my first trial, but stepped back because the number is used for specific distributions. Probably I was over-thinking. Let me give it a try again.